ddtrace/writer.py

```diff
@@ -83,44 +83,50 @@ def flush_queue(self):
             traces = self._apply_filters(traces)
         except Exception as err:
             log.error('error while filtering traces: {0}'.format(err))
+            return
 
-        traces_response = None
-
-        if traces:
-            # If we have data, let's try to send it.
-            try:
-                traces_response = self.api.send_traces(traces)
-            except Exception as err:
-                log.error('cannot send spans to {1}:{2}: {0}'.format(
-                    err, self.api.hostname, self.api.port))
+        if not traces:
+            return
 
-        if self._priority_sampler and traces_response:
-            result_traces_json = traces_response.get_json()
-            if result_traces_json and 'rate_by_service' in result_traces_json:
-                self._priority_sampler.set_sample_rate_by_service(result_traces_json['rate_by_service'])
+        # If we have data, let's try to send it.
+        traces_responses = self.api.send_traces(traces)
+        for response in traces_responses:
+            if isinstance(response, Exception) or response.status >= 400:
+                self._log_error_status(response)
```

Comment:
What if we have an `IOError`, it won't have `response.status` right?

Commenting here for inside of `self._log_error_status`, L94 here should be fine.

Looks like we did an `isinstance(response, api.Response)` up front previously.

Reply:
Yeah I missed that. Fixed.